### ISSUE TITLE
[bug] Error when build for macOS: Receiver type 'SDLOpenGLContext' for instance message is a forward declaration

### ISSUE THREAD

**Issue by paulocoutinhox** [libsdl-org/SDL#6226]
Hi,

Im trying build for macOS, but get this error.

Error:
```
receiver type 'SDLOpenGLContext' for instance message is a forward declaration
```

Full log:
https://github.com/nativium/nativium-game/actions/runs/2989300261/jobs/4792920808#step:14:4556

Can anyone help me?

**Comment by paulocoutinhox** [comment]
Any news about this?

**Comment by paulocoutinhox** [comment]
Are three lines to check:

- https://github.com/libsdl-org/SDL/blob/main/src/video/cocoa/SDL_cocoawindow.m#L280
- https://github.com/libsdl-org/SDL/blob/main/src/video/cocoa/SDL_cocoawindow.m#L282
- https://github.com/libsdl-org/SDL/blob/main/src/video/cocoa/SDL_cocoawindow.m#L2335

```
note: forward declaration of class here
@class SDLOpenGLContext;
```

```
note: forward declaration of class here
@class SDLOpenGLContext;
```

```
note: forward declaration of class here
@class SDLOpenGLContext;
```

**Comment by paulocoutinhox** [comment]
I tried compile SDL locally, and it work.

```
# library
rm -rf build
mkdir build

cmake -S . -B build -G Ninja \
    -DSDL_TESTS=ON \
    -DSDL_INSTALL_TESTS=ON \
    -DCMAKE_INSTALL_PREFIX=cmake_prefix \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_OSX_ARCHITECTURES="x86_64;arm64"

cmake --build build/ --config Release --verbose --parallel
```

Anyone knows what can be wrong?

**Comment by sezero** [comment]
Maybe @slime73 has an understanding of what's going on.

**Comment by slime73** [comment]
I'm not too familiar with using CMake to build a static SDL library on macOS.

**Comment by paulocoutinhox** [comment]
I checked and see that im disabling opengl in my example.

Maybe it is affecting, because when enable again it works.

I will closing for now because it works on macOS when i enable opengl again.

**Comment by benoit-pierre** [comment]
This is still an issue with 2.30.5, when trying to compile with `-DSDL_OPENGL=OFF -DSDL_OPENGLES=OFF`.

**Comment by slouken** [comment]
This is fixed for 2.30.6, thanks!

**Comment by benoit-pierre** [comment]
Thanks!
